### ISSUE TITLE
#[derive(Debug)] fails when used with a raw pointer to DST

### ISSUE THREAD

**Issue by whitequark** [rust-lang/rust#36870]
... with the following error message:
```
error[E0277]: the trait bound `core::ops::Fn() -> bool + 'static: core::marker::Sized` is not satisfied
   --> /home/whitequark/Work/artiq-dev/artiq/artiq/runtime/../runtime.rs/src/sched.rs:109:16
    |
109 |     Completion(*const (Fn() -> bool + 'static)),
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/whitequark/Work/artiq-dev/artiq/artiq/runtime/../runtime.rs/src/sched.rs:107:10: 107:15 note: in this expansion of #[derive(Debug)] (defined in /home/whitequark/Work/artiq-dev/artiq/artiq/runtime/../runtime.rs/src/sched.rs)
    |
    = note: `core::ops::Fn() -> bool + 'static` does not have a constant size known at compile-time
    = note: required because of the requirements on the impl of `core::fmt::Debug` for `*const core::ops::Fn() -> bool + 'static`
    = note: required because of the requirements on the impl of `core::fmt::Debug` for `&*const core::ops::Fn() -> bool + 'static`
    = note: required for the cast to the object type `core::fmt::Debug`
```

since the Debug impl for raw pointers just prints the actual pointer, I don't see why it should require the pointee to be Sized.

**Comment by apasel422** [comment]
Do you have a code snippet that exhibits this?

**Comment by apasel422** [comment]
Seems like this is a duplicate of #30427.

**Comment by whitequark** [comment]
```
trait T {}
#[derive(Debug)]
struct S(*const T);
```

**Comment by durka** [comment]
I'm working on this.

**Comment by durka** [comment]
It is not a duplicate of #30427. That was fixed by #31479 which extended the `Pointer` implementation to unsized `T`. However, the `Debug` impl, which merely delegates to `Pointer`, was not updated in that PR.

**Comment by whitequark** [comment]
Thanks @durka
